spec/factories/courses.rb

```diff
@@ -33,16 +33,16 @@
 
     study_mode { :full_time }
     resulting_in_pgce_with_qts
-
     transient do
-      subject_count      { 1 }
-      subjects           { build_list(:subject, subject_count) }
-      with_site_statuses { [] }
       with_enrichments   { [] }
       age                { nil }
       enrichments        { [] }
     end
 
+    trait :with_subject do
```

Comment:
feels a bit over the top

may have that

`subjects { [build(:subject)] }` as the default and callee can override it with blank if not needed

Reply:
This may have come from me ... I was saying that our design philosophy should be that only those associations that are _required_ by an object should be created in the factory. Anything else should be expressed explicitly. Can a `course` object exist with a `subject`?

Reply:
So what's come out of this is that a `course` should not exist with a `subject`. I say we change this here when that there changes in the model.